spec/models/package_manager/pypi_spec.rb

```diff
@@ -238,4 +238,33 @@
       end
     end
   end
+
+  describe "#mapping" do
+    let(:result) { described_class.mapping(raw_project) }
+
+    let(:raw_project) do
+      {
+        "info" => {
+          "name" => "name",
+          "summary" => "summary",
+          "home_page" => "home_page",
+          "keywords" => "keywords",
+          "classifiers" => [],
+          "project_urls" => project_urls,
+        },
+      }
+    end
+
+    context "respository url" do
+      context "project_urls.Code" do
+        let(:project_urls) do
```

Comment:
Worth adding a case for when there are no project_urls to make sure it doesn't error out? And/or a case above/below it in the ordering of the extracted data to make sure that the repository_url gets the correct one? e.g. if  project_urls contain Repository and Code, that the Repository value is used; if it only contains Code and Homepage, that Code is being used.